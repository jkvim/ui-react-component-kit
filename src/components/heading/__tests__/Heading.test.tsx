import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Heading } from '../';

describe('Heading', () => {

  describe('props', () => {

    const wrapper = mount(
      <Heading level={1}>heading</Heading>);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `level` in props', () => {
      expect(wrapper.props().level).toBe(1);
    });
  });

  it('snapshot h1', () => {
    const wrapper = mount(<Heading level={1}>heading</Heading>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot h2', () => {
    const wrapper = mount(<Heading level={2}>heading</Heading>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot h3', () => {
    const wrapper = mount(<Heading level={3}>heading</Heading>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot h4', () => {
    const wrapper = mount(<Heading level={4}>heading</Heading>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot h5', () => {
    const wrapper = mount(<Heading level={5}>heading</Heading>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot h6', () => {
    const wrapper = mount(<Heading level={6}>heading</Heading>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});