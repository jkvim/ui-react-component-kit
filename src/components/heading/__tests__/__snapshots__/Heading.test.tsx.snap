// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading props should match snapshot 1`] = `
<Heading
  level={1}
>
  <styled.h1
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "2.8rem",
            "h2": "2.1rem",
            "h3": "1.6rem",
            "h4": "1.3rem",
            "h5": "1.2rem",
            "h6": "1.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <h1
      className="sc-bdVaJa bSTzsa"
    >
      heading
    </h1>
  </styled.h1>
</Heading>
`;

exports[`Heading snapshot h1 1`] = `
<Heading
  level={1}
>
  <styled.h1
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "2.8rem",
            "h2": "2.1rem",
            "h3": "1.6rem",
            "h4": "1.3rem",
            "h5": "1.2rem",
            "h6": "1.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <h1
      className="sc-bdVaJa bSTzsa"
    >
      heading
    </h1>
  </styled.h1>
</Heading>
`;

exports[`Heading snapshot h2 1`] = `
<Heading
  level={2}
>
  <styled.h2
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "2.8rem",
            "h2": "2.1rem",
            "h3": "1.6rem",
            "h4": "1.3rem",
            "h5": "1.2rem",
            "h6": "1.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <h2
      className="sc-bwzfXH fCLLGB"
    >
      heading
    </h2>
  </styled.h2>
</Heading>
`;

exports[`Heading snapshot h3 1`] = `
<Heading
  level={3}
>
  <styled.h3
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "2.8rem",
            "h2": "2.1rem",
            "h3": "1.6rem",
            "h4": "1.3rem",
            "h5": "1.2rem",
            "h6": "1.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <h3
      className="sc-htpNat BbwKb"
    >
      heading
    </h3>
  </styled.h3>
</Heading>
`;

exports[`Heading snapshot h4 1`] = `
<Heading
  level={4}
>
  <styled.h4
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "2.8rem",
            "h2": "2.1rem",
            "h3": "1.6rem",
            "h4": "1.3rem",
            "h5": "1.2rem",
            "h6": "1.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <h4
      className="sc-bxivhb UWGzl"
    >
      heading
    </h4>
  </styled.h4>
</Heading>
`;

exports[`Heading snapshot h5 1`] = `
<Heading
  level={5}
>
  <styled.h5
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "2.8rem",
            "h2": "2.1rem",
            "h3": "1.6rem",
            "h4": "1.3rem",
            "h5": "1.2rem",
            "h6": "1.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <h5
      className="sc-ifAKCX dBEikb"
    >
      heading
    </h5>
  </styled.h5>
</Heading>
`;

exports[`Heading snapshot h6 1`] = `
<Heading
  level={6}
>
  <styled.h6
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
            "h1": "2.8rem",
            "h2": "2.1rem",
            "h3": "1.6rem",
            "h4": "1.3rem",
            "h5": "1.2rem",
            "h6": "1.1rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <h6
      className="sc-EHOje hpwIWC"
    >
      heading
    </h6>
  </styled.h6>
</Heading>
`;
